test/models-resolve.test.js

```diff
@@ -0,0 +1,141 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: loopback-connector-remote
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const assert = require('assert');
+const helper = require('./helper');
+const loopback = require('loopback');
+const sinon = require('sinon');
+
+const RemoteConnector = require('../lib/remote-connector');
+
+describe('Model Resolve Tests', function() {
+  let serverApp, clientApp, remoteDs, resolveSpy, ChildModel;
+
+  before('spy remote connector resolve() function', () => {
+    resolveSpy = sinon.spy(RemoteConnector.prototype, 'resolve');
+  });
+
+  beforeEach('reset the resolve()', () => {
+    resolveSpy.reset();
+  });
+
+  after('restore remote connector resolve() function', () => {
+    resolveSpy.restore();
+  });
+
+  before('create remote datasource', () => {
+    serverApp = helper.createRestAppAndListen();
+    clientApp = loopback({localRegistry: true});
+    remoteDs = helper.createRemoteDataSource(clientApp, serverApp);
+  });
+
+  before('create a child model', () => {
+    const db = helper.createMemoryDataSource(clientApp);
+    ChildModel = clientApp.registry.createModel({
+      name: 'ChildModel',
+    });
+    clientApp.model(ChildModel, {dataSource: db});
+  });
+
+  it('should resolve a remote model only once (hasMany)', () => {
+    const RemoteModel = clientApp.registry.createModel({
+      name: 'RemoteModelHasMany',
+      relations: {
+        children: {
+          type: 'hasMany',
+          model: 'ChildModel',
+        },
+      },
+    });
+    clientApp.model(RemoteModel, {dataSource: remoteDs});
+    assert(resolveSpy.withArgs(RemoteModel).calledOnce);
```

Comment:
Please use [Sinon assertions](http://sinonjs.org/releases/v4.1.3/assertions/).

`assert(<code returning true or false>)` produces failure messages that are difficult to troubleshoot - they only say that the test failed, but don't provide any additional insights about what went wrong.

I believe Sinon Asserts provide better error messages.
  
Please all fix other newly added tests in a similar way.